app/models/measurement.rb

```diff
@@ -16,23 +18,17 @@ class Measurement < ApplicationRecord
     FAMILY_NAME: "family_name"
   }.freeze
 
-  # delegate :tank, to: :measurement_event
+  # The below code is unstable. This is retrofitting the values from the seeded CSV
+  # In reality, for instance, the subject would not always be a tank
   def self.create_from_csv_data(attrs)
     # remove relational (non-attribute) data from hash to be handled separately
     measurement_event_name = attrs.fetch(:measurement_event)
-    if attrs[:tank_name]
-      tank = Tank.find_or_create_by!(
-        name: attrs.fetch(:tank_name),
-        organization_id: attrs.fetch(:organization_id)
-      )
-    end
-    if attrs[:animal_pii_tag]
-      animal = Animal.find_or_create_by!(
-        pii_tag: attrs.fetch(:animal_pii_tag),
-        organization_id: attrs.fetch(:organization_id)
-      )
-    end
-    family = Family.find_by!(name: attrs.fetch(:family_name)) if attrs[:family_name]
+    tank = Tank.find_or_create_by!(
+      name: attrs.fetch(:tank_name),
+      organization_id: attrs.fetch(:organization_id)
+    )
+
+    measurement_type = MeasurementType.find_or_create_by!(name: "length", unit: "cm", organization_id: attrs.fetch(:organization_id))
```

Comment:
I'm having a little trouble understanding this - it looks like it hardcodes the incorrect measurement type. Was there supposed to be a follow-on commit?